### PR TITLE
Add egress firewall docs for OVN-Kubernetes [4.6]

### PR DESCRIPTION
**Pull request by jboxman**
- https://issues.redhat.com/browse/OSDOCS-976

Previews:

- [Configuring an egress firewall to control access to external IP addresses](https://OSDOCS-976--ocpdocs.netlify.com/openshift-enterprise/latest/networking/ovn_kubernetes_network_provider/configuring-egress-firewall.html)
- [Editing an egress firewall for a project](https://OSDOCS-976--ocpdocs.netlify.com/openshift-enterprise/latest/networking/ovn_kubernetes_network_provider/editing-egress-firewall.html)
- [Removing an egress firewall from a project](https://OSDOCS-976--ocpdocs.netlify.com/openshift-enterprise/latest/networking/ovn_kubernetes_network_provider/removing-egress-firewall.html)
- [Viewing an egress firewall in a project](https://OSDOCS-976--ocpdocs.netlify.com/openshift-enterprise/latest/networking/ovn_kubernetes_network_provider/viewing-egress-firewall.html)

### CODE REVIEW

**Comment by jboxman** [comment]
Hi @JacobTanenbaum,

This PR updates the egress firewall documentation to include OVN-Kubernetes support. The most relevant changes are probably (full previews in the [first comment](https://github.com/openshift/openshift-docs/pull/20719#issue-394504305)):

- https://github.com/openshift/openshift-docs/pull/20719/files#diff-8bbe4c2b7e5e8cc4e45457a480a07126R35
- https://github.com/openshift/openshift-docs/pull/20719/files#diff-65140b369f920bfd8be8c87fc1f37b43R109
- https://github.com/openshift/openshift-docs/pull/20719/files#diff-65140b369f920bfd8be8c87fc1f37b43R52
- https://github.com/openshift/openshift-docs/pull/20719/files#diff-65140b369f920bfd8be8c87fc1f37b43R65
- https://github.com/openshift/openshift-docs/pull/20719/files#diff-65140b369f920bfd8be8c87fc1f37b43R69

Can you confirm whether these changes are correct?

Thanks!

**Comment by jboxman** [comment]
@JacobTanenbaum, I've updated this to exclude DNS for OVN-Kubernetes only; Can you look and confirm whether the docs are accurate for OVN-Kubernetes?

Thanks!

**Comment by jboxman** [comment]
Hi @huiran0826,

Can you verify the changes to the egress firewall documentation for OVN-Kubernetes?

Thanks!

File: modules/nw-egressnetworkpolicy-about.adoc [hunk 1]
```diff
@@ -16,45 +26,60 @@ to internal hosts that are outside the {product-title} cluster.
...
 An egress firewall has the following limitations:
 
-* No project can have more than one EgressNetworkPolicy object.
+* No project can have more than one {kind} object.
 
-* A maximum of 1 EgressNetworkPolicy object with a maximum of 50 rules can be defined per project.
+* A maximum of 1 {Kind} object with a maximum of 50 rules can be defined per project.
 
-* The `default` project cannot use egress network policy.
+* The `default` project cannot use an egress firewall.
```
Comment (huiran0826) [inline modules/nw-egressnetworkpolicy-about.adoc:65]: EgressFirewall will work in the default project in ovn-kubernetes but not in sdn

File: modules/nw-egressnetworkpolicy-object.adoc [hunk 2]
```diff
@@ -1,70 +1,156 @@
...
+apiVersion: {api}
+kind: {kind}
 metadata:
   name: <name> <1>
 spec:
   egress: <2>
     ...
 ----
-<1> Specify a `name` for your egress firewall policy.
+<1> A name for your egress firewall policy.
```
Comment (huiran0826) [inline modules/nw-egressnetworkpolicy-object.adoc:35]: The EgressFirewall policy name only can be "default" for now, cannot use other name here,  refer to bug:https://bugzilla.redhat.com/show_bug.cgi?id=1862847

File: modules/nw-egressnetworkpolicy-object.adoc [hunk 3]
```diff
@@ -1,70 +1,156 @@
...
 The following example defines several egress firewall policy rules:
 
-[source,yaml]
+[source,yaml,subs="attributes+"]
 ----
 apiVersion: network.openshift.io/v1
-kind: EgressNetworkPolicy
+kind: {kind}
 metadata:
   name: default-rules <1>
```
Comment (huiran0826) [inline modules/nw-egressnetworkpolicy-object.adoc:105]: The name should be "default"

File: modules/nw-egressnetworkpolicy-object.adoc [hunk 4]
```diff
@@ -1,70 +1,156 @@
...
 <1> The name for the policy object.
 
 <2> A collection of egress firewall policy rule objects.
+
+ifdef::ovn[]
+The following example defines a policy rule that denies traffic to the host at the `172.16.1.1` IP address, if the traffic is using either the TCP protocol and destination port `80` or any protocol and destination port `443`.
+
+[source,yaml,subs="attributes+"]
+----
+apiVersion: network.openshift.io/v1
```
Comment (huiran0826) [inline modules/nw-egressnetworkpolicy-object.adoc:129]: Here the apiVersion should use "k8s.ovn.org/v1" which is for OVN

**Comment by jboxman** [comment]
@huiran0826 thank you!

**Comment by jboxman** [comment]
@JacobTanenbaum, can you PTAL? Thanks!

**Comment by jboxman** [comment]
@huiran0826, updated!

File: networking/ovn_kubernetes_network_provider/configuring-egress-firewall.adoc [hunk 5]
```diff
@@ -0,0 +1,12 @@
+[id="ovn-configuring-egress-firewall"]
+= Configuring an egress firewall for a project
+include::modules/common-attributes.adoc[]
+:context: ovn-configuring-egress-firewall
+
+toc::[]
+
+As a cluster administrator, you can create an egress firewall for a project that will restrict egress traffic leaving your {product-title} cluster.
```
Comment (adellape) [inline networking/ovn_kubernetes_network_provider/configuring-egress-firewall.adoc:8]: s/will restrict/restricts/

File: networking/openshift_sdn/configuring-egress-firewall.adoc [hunk 6]
```diff
@@ -1,14 +1,12 @@
 [id="configuring-egress-firewall"]
-= Configuring an egress firewall to control access to external IP addresses
+= Configuring an egress firewall for a project
 include::modules/common-attributes.adoc[]
-:context: configuring-an-egress-firewall
+:context: openshift-sdn-egress-firewall
 
 toc::[]
 
 As a cluster administrator, you can create an egress firewall for a project that will restrict egress traffic leaving your {product-title} cluster.
```
Comment (adellape) [inline networking/openshift_sdn/configuring-egress-firewall.adoc:8]: s/will restrict/restricts/

File: modules/nw-egressnetworkpolicy-create.adoc [hunk 7]
```diff
@@ -25,25 +37,34 @@ If the project already has an EgressNetworkPolicy object defined, you must edit
 policy rules.
 .. In the file you created, define an egress policy object.
 
-. Enter the following command to create the policy object:
+. Enter the following command to create the policy object. Replace `<policy-name>` with the name of the policy and `<project>` with the project that the rule applies to.
```
Comment (adellape) [inline modules/nw-egressnetworkpolicy-create.adoc:40]: Throughout, the replaceables should [use underscores](https://github.com/openshift/openshift-docs/blob/master/contributing_to_docs/doc_guidelines.adoc#code-blocks-command-syntax-and-example-output).

File: modules/nw-egressnetworkpolicy-view.adoc [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+
+. To inspect a policy, enter the following command. Replace `<policy-name>` with the name of the policy to inspect.
++
+[source,terminal,subs="attributes+"]
+----
+$ oc describe {res} <policy-name>
+----
++
+.Example output
+----
```
Comment (adellape) [inline modules/nw-egressnetworkpolicy-view.adoc:46]: `[source,terminal]` here?

**Review by adellape — APPROVED** [review]
LGTM! Just a few nits.

**Comment by jboxman** [comment]
@adellape, thanks!

**Comment by huiran0826** [comment]
@jboxman  After create the egressfirewall, the output should be "egressfirewall.k8s.ovn.org/default created
"  ,  not "egressfirewall.network.openshift.io/default created"

**Pull request merged**